src/ruby/puppetserver-lib/puppet/server/catalog.rb

```diff
@@ -0,0 +1,121 @@
+require 'puppet/server'
+
+module Puppet::Server::Catalog
+
+  def self.convert_java_args_to_ruby(hash)
+    Hash[hash.collect do |key, value|
+        # Stolen and modified from params_to_ruby in handler.rb
+        newkey = key.to_s
+        # Java::ClojureLang::Keywords retain their leading colon when
+        # converted to a string
+        newkey.slice!(0)
```

Comment:
I get what's happening here, though using Ruby's convenience methods might make it more readable:
`newkey.delete_prefix!(':')`